src/test/java/org/jenkinsci/plugins/badge/dsl/AddEmbeddableBadgeConfigStepTest.java

```diff
@@ -1,28 +1,64 @@
+        assertEquals("testId", step.getID());
+    }
+
+    @Test
+    void testGetSubjectAndSetSubject() {
+        AddEmbeddableBadgeConfigStep step = new AddEmbeddableBadgeConfigStep("testId");
+        assertNull(step.getSubject());
+        step.setSubject("testSubject");
+        assertEquals("testSubject", step.getSubject());
+    }
+
+    @Test
+    void testGetStatusAndSetStatus() {
+        AddEmbeddableBadgeConfigStep step = new AddEmbeddableBadgeConfigStep("testId");
+        assertNull(step.getStatus());
+        step.setStatus("testStatus");
+        assertEquals("testStatus", step.getStatus());
+    }
+
+    @Test
+    void testGetColorAndSetColor() {
+        AddEmbeddableBadgeConfigStep step = new AddEmbeddableBadgeConfigStep("testId");
+        assertNull(step.getColor());
+        step.setColor("testColor");
+        assertEquals("testColor", step.getColor());
+    }
+
+    @Test
+    void testGetAnimatedOverlayColorAndSetAnimatedOverlayColor() {
+        AddEmbeddableBadgeConfigStep step = new AddEmbeddableBadgeConfigStep("testId");
+        assertNull(step.getAnimatedOverlayColor());
+        step.setAnimatedOverlayColor("testOverlayColor");
+        assertEquals("testOverlayColor", step.getAnimatedOverlayColor());
+    }
 
-public class AddEmbeddableBadgeConfigStepTest {
     @Test
-    public void testConstructor() {
-        AddEmbeddableBadgeConfigStep addEmbeddableBadgeConfigStep =
-                new AddEmbeddableBadgeConfigStep("test-Id-constructor");
-        assertThat(addEmbeddableBadgeConfigStep.getID(), is("test-Id-constructor"));
-        assertThat(addEmbeddableBadgeConfigStep.getSubject(), is(nullValue()));
-        assertThat(addEmbeddableBadgeConfigStep.getStatus(), is(nullValue()));
-        assertThat(addEmbeddableBadgeConfigStep.getColor(), is(nullValue()));
-        assertThat(addEmbeddableBadgeConfigStep.getAnimatedOverlayColor(), is(nullValue()));
-        assertThat(addEmbeddableBadgeConfigStep.getLink(), is(nullValue()));
+    void testGetLinkAndSetLink() {
+        AddEmbeddableBadgeConfigStep step = new AddEmbeddableBadgeConfigStep("testId");
+        assertNull(step.getLink());
+        step.setLink("testLink");
+        assertEquals("testLink", step.getLink());
     }
 
     @Test
-    public void testGetAnimatedOverlayColor() {
-        AddEmbeddableBadgeConfigStep addEmbeddableBadgeConfigStep =
-                new AddEmbeddableBadgeConfigStep("test-animated-overlay-color");
-        assertThat(addEmbeddableBadgeConfigStep.getColor(), is(nullValue()));
+    void testDescriptor() {
+        AddEmbeddableBadgeConfigStep.DescriptorImpl descriptor = new AddEmbeddableBadgeConfigStep.DescriptorImpl();
+        assertEquals("addEmbeddableBadgeConfiguration", descriptor.getFunctionName());
+        assertEquals("Add an Embeddable Badge Configuration", descriptor.getDisplayName());
+        assertEquals(1, descriptor.getRequiredContext().size());
```

Comment:
Optional: A Hamcrest assertion is easier to read when it fails.  In this case, the `getRequiredContext()` return value would be shown to the user when it fails.
```suggestion
        assertEquals(1, descriptor.getRequiredContext().size());
```